cmd/deploy.go

```diff
@@ -180,7 +180,7 @@ func autofillApimodel(dc *deployCmd) {
 
 	if !useManagedIdentity {
 		spp := dc.containerService.Properties.ServicePrincipalProfile
-		if spp != nil && (spp.ClientID == "" || spp.Secret == "") {
+		if spp != nil && spp.KeyvaultSecretRef == "" && (spp.ClientID == "" || spp.Secret == "") {
```

Comment:
the code in deploy.go is to conveniently create spn if it's not specified. this will only be applicable when keyvault secret ref is not used.